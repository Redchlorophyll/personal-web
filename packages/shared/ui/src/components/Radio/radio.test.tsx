import React, { useState } from 'react';
import '@testing-library/jest-dom';
import { Radio } from './index';
import {
  renderWithProviders,
  screen,
  userEvent,
  act,
} from 'testing-library-react-custom';

function RadioWrapper() {
  const [radio, setRadio] = useState<string>('');

  function onChangeCheckbox(val: string) {
    setRadio(val);
  }

  return (
    <div>
      {radio}
      <Radio
        valueGroup={radio}
        onChange={(val: string) => onChangeCheckbox(val)}
        value={'test'}
      />
    </div>
  );
}

describe('shared - ui - radio', () => {
  test('should render properly', () => {
    const { baseElement } = renderWithProviders(<RadioWrapper />);

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  test('it should render properly', () => {
    renderWithProviders(<RadioWrapper />);
    const target = screen.getByRole('radio');
    expect(target).toBeInTheDocument();
  });

  test('it should be checked when clicked', async () => {
    const user = userEvent.setup();
    renderWithProviders(<RadioWrapper />);
    const target = screen.getByRole('radio');
    expect(screen.queryByText('test')).not.toBeInTheDocument();

    await act(async () => await user.click(target));
    expect(screen.queryByText('test')).toBeInTheDocument();

    await act(async () => await user.click(target));
    expect(screen.queryByText('test')).not.toBeInTheDocument();
  });
});
