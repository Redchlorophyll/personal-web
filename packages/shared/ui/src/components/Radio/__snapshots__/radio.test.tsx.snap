// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared - ui - radio should render properly 1`] = `
<body>
  <div>
    <div>
      <label>
        <input
          class="w-6 h-6 peer sr-only"
          type="radio"
          value="test"
        />
        <div
          class="inline-block peer-checked:[&>*]:bg-primary-700 w-6 h-6 bg-black-100 peer-checked:bg-transparent rounded-full p-[1.1px] border-solid border-[1px] border-black-800"
        >
          <div
            class="w-5 h-5 rounded-full"
          />
        </div>
      </label>
    </div>
  </div>
</body>
`;
